src/components/UserSelector.tsx

```diff
@@ -1,19 +1,69 @@
-import React from 'react';
+import React, {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+
+type Props = {
+  users: User[] | null | undefined
+  setDropDownOpen: Dispatch<SetStateAction<boolean>>;
+  dropDownOpen: boolean;
+  setUserSelectedId: Dispatch<SetStateAction<number>>;
+  userSelectedId: number;
+  setPostSelectedId: Dispatch<SetStateAction<number>>
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  setDropDownOpen,
+  dropDownOpen,
+  setUserSelectedId,
+  userSelectedId,
+  setPostSelectedId,
+}) => {
+  const [selectedUser, setSelectedUser] = useState<User>();
+  const showOptions = () => {
+    setDropDownOpen(prevState => !prevState);
+  };
+
+  const handleUserKeyPress = useCallback(event => {
+    const dropDown = document.getElementById('dropdown-menu');
```

Comment:
You can use refs for manipulation with DOM elements. Read about useRef hook

Reply:
Done